java-api-core/src/com/kinvey/java/query/AbstractQuery.java

```diff
@@ -102,6 +102,10 @@ public String getQueryFilterJson(JsonFactory factory) {
             jsonResult = writer.toString();
         } catch (Exception ex) {}
 
+        if (jsonResult.equals("{}")) {
```

Comment:
It's need because this case doesn't depend on the case above. When we set it to null then the empty query won't be added.